core/sql/arkcmp/CmpStatement.cpp

```diff
@@ -1072,7 +1082,8 @@ CmpStatement::process(const CmpMessageDDLwithStatus &statement)
 				 odbcProcess,
 				 noTextCache,
 				 aqrPrepare,
-				 standaloneQuery))
+				 standaloneQuery,
+				 doNotCachePlan))
```

Comment:
Yes. In the separate compiler case, the CmpCompileInfo object is sent to the tdm_arkcmp process. In the embedded case, it is passed to CmpContext::compileDirect. Either way, the bit travels to CmpStatement::process and ultimately down to CmpMain::sqlcomp. I've unit tested both paths.